litestar/dto/factory/backends/msgspec/backend.py

```diff
@@ -32,6 +32,9 @@ class MsgspecDTOBackend(AbstractDTOBackend[Struct]):
     def parse_raw(self, raw: bytes, media_type: MediaType | str) -> Struct | Collection[Struct]:
         return decode_media_type(raw, media_type, type_=self.annotation)  # type:ignore[no-any-return]
 
+    def populate_data_from_builtins(self, model_type: type[T], data: Any) -> T | Collection[T]:
+        raise NotImplementedError("Msgspec backend does not support marshalling types")
```

Comment:
It does support this via `from_builtins`. Maybe a different error message would be appropriate here?

Reply:
Oh. I assumed from the conversation this morning that there was no coercion of types, but appears that's not the case.

I'm a bit confused with all this now - do I even need the pydantic backend then? I'll experiment with `from_builtins()` tomorrow. Thanks for the heads-up.

Reply:
It does, but it's not exactly equivalent to what Pydantic does. I believe the issue was with the passthrough of unsupported types. 

https://jcristharif.com/msgspec/converters.html

Reply:
I ended up inmplementing this on the msgspec backed, but still using the pydantic backend for form data for now.